githubauth/app.go

```diff
@@ -89,21 +88,39 @@ func NewApp[T *rsa.PrivateKey | string | []byte](appID string, privateKeyT T, op
 		privateKey = t
 	case string:
 		privateKey, err = parseRSAPrivateKeyPEM([]byte(t))
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse private key as a PEM-encoded string: %w", err)
-		}
 	case []byte:
 		privateKey, err = parseRSAPrivateKeyPEM(t)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse private key as a PEM-encoded []byte: %w", err)
-		}
 	default:
 		panic("impossible")
 	}
+	return func(g *App) (*App, error) {
+		if err != nil {
+			return nil, err
+		}
+		g.signer = NewPrivateKeySigner(privateKey)
+		return g, nil
+	}
+}
+
+// WithKMSSigner is one of two options that configures how requests
+// to GitHub are signed. This option utilizes Google Cloud KMS
+// via the provided keyID to sign the request. The keyID is in the
+// format "projects/*/locations/*/keyRings/*/cryptoKeys/*.".
+func WithKMSSigner(ctx context.Context, keyID string) Option {
```

Comment:
Might want to say "GoogleCloudKMS" in case we need to support AWS KMS or Vault KMS in the future.